src/studio/src/designer/frontend/shared/features/dataModelling/DataModelling.tsx

```diff
@@ -28,6 +74,12 @@ type shouldSelectFirstEntryProps = {
   metadataLoadingState: LoadingState;
 };
 
+enum LandingDialogState {
+  HAS_NOT_YET_BEEN_SHOWN, // Datamodels are not yet loaded
+  IS_VISIBLE, // Dialog should be visible
+  IS_NOT_GOING_TO_BE_SHOWN // Dialog should never (again) be visible
```

Comment:
Would be good to use string enums instead of numbers. With numbers, the order may be important, and someone could easily make the mistake of adding a new entry to the beginning shifting all the values by 1.

Also could the names perhaps be modified a bit so we wont need the comments?

```suggestion
  DatamodelsNotLoaded: 'DatamodelsNotLoaded',
  DialogIsVisible: 'DialogIsVisible',
  DialogShouldNotBeShown: 'DialogShouldNotBeShown',
```